Adds diagrams and updates doc (#139)
dedis/dela: docs/architecture.md

# Architecture

Dela has been built with modularization and testability in mind. As such, the

> system is made of multiple interoperable modules' implementations based on a set

Reviewer: ```suggestion
system is made of multiple interoperable modules based on a set
```

of well-defined abstractions.

## Stacks

The stacks display the role of each module and its relation to others. Each

> module can rely on zero or more of the modules bellow it. From the client's

Reviewer: ```suggestion
module can rely on zero or more of the modules below it. From the client's
```

perspective the stack is simpler since it doesn't need to know about the complex
ordering mechanism.

![stacks](assets/stacks.png)

## Execution flow

> The execution flow displays how the system uses the modules in order to fullfil

Reviewer: ```suggestion
The execution flow displays how the system uses the modules in order to fulfill
```
